### ISSUE TITLE
Error when using useApolloClient ("No Apollo Client instance can be found")

### ISSUE THREAD

**Issue by krinoid** [redwoodjs/redwood#2886]
Hi, I've been using redwood for a couple of months and really enjoy it, thanks to all contributors!

Recently, I've been working on updating the project I have on `v0.31.2` to the latest one (`v0.34`). During that I've encountered an error while using `useApolloClient` in some of my pages/components. The error message is: "No Apollo Client instance can be found. Please ensure that you have called `ApolloProvider` higher up in your tree.".

This issue is reproduced here: https://github.com/krinoid/redwood-useApolloClient-bug-reproduction
Steps:
1. Clone & install dependencies
2. Open http://localhost:8910/test
3. You should see the error:
<img width="1091" alt="image" src="https://user-images.githubusercontent.com/25506297/123271101-c8c4f780-d500-11eb-9c9d-4a6697e7ddc4.png">

The code that causes it:
```jsx
import { Link, routes } from '@redwoodjs/router'
import { useApolloClient } from '@apollo/client'

const TestPage = () => {
  const apolloClient = useApolloClient()
  return (
    <>
      <h1>TestPage</h1>
      <p>
        Find me in <code>./web/src/pages/TestPage/TestPage.js</code>
      </p>
      <p>
        My default route is named <code>test</code>, link to me with `
        <Link to={routes.test()}>Test</Link>`
      </p>
    </>
  )
}

export default TestPage
```

To be clear, `App.js` includes `RedwoodApolloProvider`:
```jsx
import { FatalErrorBoundary } from '@redwoodjs/web'
import { RedwoodApolloProvider } from '@redwoodjs/web/apollo'

import FatalErrorPage from 'src/pages/FatalErrorPage'
import Routes from 'src/Routes'

import './index.css'

const App = () => (
  <FatalErrorBoundary page={FatalErrorPage}>
    <RedwoodApolloProvider>
      <Routes />
    </RedwoodApolloProvider>
  </FatalErrorBoundary>
)

export default App
```

I'm still investigating what might be causing the problem but decided to share info about the issue before finding the source, maybe somebody will figure it out sooner.

**Comment by krinoid** [comment]
OK, I figured out that the problem was caused by the `@apollo/client` version mismatch between what `@redwood/web` uses internally (`3.3.19`) and latest version installed by `yarn add @apollo/client` (`3.3.20`). Pinning `@apollo/client` in my project to `3.3.19` (note the lack of `^`) solved the problem. The real root cause is happening deeper down the dependencies tree, I guess but my first priority was ability to update to the latest version of redwood. For comparison/reference, I've pushed a branch with changes that fix the immediate issue: https://github.com/krinoid/redwood-useApolloClient-bug-reproduction/compare/fix/useApolloClient-version-fix

I'll keep the issue open, so that maintainers can see it -- feel free to close it after that.
